File: internal/dto/dist_order.go
package dto

type DistOrderItemInput struct {
	SkuID           uint64  `json:"skuId"`
	OfferID         uint64  `json:"offerId"`
	ProductName     string  `json:"productName"`
	SkuCode         string  `json:"skuCode"`
	SkuSpecs        string  `json:"skuSpecs"`
	PicURL          string  `json:"picUrl"`
	DistributorSkuCode string  `json:"distributorSkuCode"`
	Qty             int     `json:"qty" binding:"required,min=1"`
	SaleUnitPrice   float64 `json:"saleUnitPrice"`
	SaleAmount      float64 `json:"saleAmount"`
	UnitPrice       float64 `json:"unitPrice"`
	RefSoID         uint64  `json:"refSoId"`
	RefOrderNo      string  `json:"refOrderNo"`
	Remark          string  `json:"remark"`
}

type DistOrderInput struct {
	DistributorID          uint64                   `json:"distributorId" binding:"required"`
	FulfillmentType     string                   `json:"fulfillmentType"`
	Currency            string                   `json:"currency"`
	ExpectedArrivalDate string                   `json:"expectedArrivalDate"`
	WarehouseID         uint64                   `json:"warehouseId"`
	RefSoID             uint64                   `json:"refSoId"`
	RefTraceID          string                   `json:"refTraceId"`
	OrderedAt           string                   `json:"orderedAt"` // 采购时间；手工新建可改，默认当天
	SaleAmount          float64                  `json:"saleAmount"` // 销售侧订单总实付
	Remark              string                   `json:"remark"`
	Items               []DistOrderItemInput `json:"items" binding:"required,min=1,dive"`
}

type DistOrderItemDetail struct {
	ID              uint64  `json:"id"`
	SkuID           uint64  `json:"skuId"`
	OfferID         uint64  `json:"offerId"`
	ProductName     string  `json:"productName"`
	SkuCode         string  `json:"skuCode"`
	SkuSpecs        string  `json:"skuSpecs"`
	PicURL          string  `json:"picUrl"`
	DistributorSkuCode string  `json:"distributorSkuCode"`
	Qty             int     `json:"qty"`
	SaleUnitPrice   float64 `json:"saleUnitPrice"`
	SaleAmount      float64 `json:"saleAmount"`
	UnitPrice       float64 `json:"unitPrice"`
	LineAmount      float64 `json:"lineAmount"`
	ReceivedQty     int     `json:"receivedQty"`
	RefSoID         uint64  `json:"refSoId,omitempty"`
	RefOrderNo      string  `json:"refOrderNo,omitempty"`
	Cancelled       bool    `json:"cancelled"`
	Remark          string  `json:"remark"`
}

type DistOrderDetail struct {
	ID                  uint64                    `json:"id"`
	DistNo                string                    `json:"distNo"`
	DistributorID          uint64                    `json:"distributorId"`
	DistributorName        string                    `json:"distributorName"`
	DistributorCode        string                    `json:"distributorCode"`
	Status              string                    `json:"status"`
	TotalAmount         float64                   `json:"totalAmount"`
	SaleAmount          float64                   `json:"saleAmount"`
	Currency            string                    `json:"currency"`
	ExpectedArrivalDate string                    `json:"expectedArrivalDate,omitempty"`
	WarehouseID         uint64                    `json:"warehouseId"`
	FulfillmentType     string                    `json:"fulfillmentType"`
	RefSoID             uint64                    `json:"refSoId"`
	RefTraceID          string                    `json:"refTraceId"`
	BuyerID             uint64                    `json:"buyerId"`
	BuyerName           string                    `json:"buyerName"`
	PayStatus           string                    `json:"payStatus"`
	Remark              string                    `json:"remark"`
	OrderedAt           string                    `json:"orderedAt,omitempty"`
	CompletedAt         string                    `json:"completedAt,omitempty"`
	CreatedAt           string                    `json:"createdAt"`
	Items               []DistOrderItemDetail `json:"items"`
}

type DistOrderListItem struct {
	ID              uint64  `json:"id"`
	DistNo            string  `json:"distNo"`
	DistributorID      uint64  `json:"distributorId"`
	DistributorName    string  `json:"distributorName"`
	Status          string  `json:"status"`
	PayStatus       string  `json:"payStatus"`
	FulfillmentType string  `json:"fulfillmentType"`
	TotalAmount     float64 `json:"totalAmount"`
	SaleAmount      float64 `json:"saleAmount"`
	Currency        string  `json:"currency"`
	ItemCount       int     `json:"itemCount"`
	SkuSpecs        string  `json:"skuSpecs"` // 明细规格汇总（同规格累加数量，如「规格 x2」；分号分隔）
	RefSoID         uint64  `json:"refSoId,omitempty"`
	RefTraceID      string  `json:"refTraceId,omitempty"`
	OrderedAt       string  `json:"orderedAt,omitempty"`
	CreatedAt       string  `json:"createdAt"`
}

type MergeDistOrdersInput struct {
	SourceDistOrderIDs []uint64 `json:"sourceDistOrderIds" binding:"required,min=2"`
	TargetDistOrderID  uint64   `json:"targetDistOrderId"` // 可选；默认取 sourceDistOrderIds[0]
}

type MergeDistOrdersResult struct {
	*DistOrderDetail
	MergedFromDistNos []string `json:"mergedFromDistNos"`
	Relinked        int64    `json:"relinked"`
}

// DetachSalesOrderInput 从代发单中撤回某笔销售单（明细标灰作废 + 备注说明）。
type DetachSalesOrderInput struct {
	DistNo     string `json:"distNo"`
	OrderNo  string `json:"orderNo"`
	SoID     uint64 `json:"soId"`
	Reason   string `json:"reason"`
}

// UpdateDistOrderItemPriceInput 更新采购明细单价（已下单未付款也可改）。
type UpdateDistOrderItemPriceInput struct {
	ItemID    uint64  `json:"itemId" binding:"required"`
	UnitPrice float64 `json:"unitPrice"`
}

type UpdateDistOrderItemPricesInput struct {
	Items []UpdateDistOrderItemPriceInput `json:"items" binding:"required,min=1,dive"`
}

File: web/src/api/distOrder.ts
import client, { unwrap, type PageData } from './client'

export interface DistOrderItem {
  id?: number
  skuId: number
  offerId?: number
  productName?: string
  skuCode?: string
  skuSpecs?: string
  picUrl?: string
  distributorSkuCode?: string
  qty: number
  saleUnitPrice?: number
  saleAmount?: number
  unitPrice: number
  lineAmount?: number
  receivedQty?: number
  refSoId?: number
  refOrderNo?: string
  cancelled?: boolean
  remark?: string
}

export interface DistOrder {
  id: number
  distNo: string
  distributorId: number
  distributorName?: string
  distributorCode?: string
  status: string
  totalAmount: number
  saleAmount?: number
  currency: string
  expectedArrivalDate?: string
  warehouseId?: number
  fulfillmentType: string
  refSoId?: number
  refTraceId?: string
  buyerId?: number
  buyerName?: string
  payStatus: string
  remark?: string
  orderedAt?: string
  completedAt?: string
  createdAt?: string
  items: DistOrderItem[]
}

export interface DistOrderListItem {
  id: number
  distNo: string
  distributorId: number
  distributorName?: string
  status: string
  payStatus: string
  fulfillmentType: string
  totalAmount: number
  currency: string
  itemCount: number
  skuSpecs?: string
  refSoId?: number
  refTraceId?: string
  orderedAt?: string
  createdAt: string
}

export interface DistOrderInput {
  distributorId?: number
  fulfillmentType?: string
  currency?: string
  expectedArrivalDate?: string
  warehouseId?: number
  refSoId?: number
  refTraceId?: string
  orderedAt?: string
  saleAmount?: number
  remark?: string
  items: Array<{
    skuId?: number
    offerId?: number
    productName?: string
    skuCode?: string
    skuSpecs?: string
    picUrl?: string
    distributorSkuCode?: string
    qty: number
    saleUnitPrice?: number
    saleAmount?: number
    unitPrice?: number
    remark?: string
  }>
}

export const DIST_STATUS_MAP: Record<string, { label: string; type: '' | 'success' | 'warning' | 'info' | 'danger' }> = {
  draft: { label: '草稿', type: 'info' },
  confirmed: { label: '已确认', type: '' },
  paid: { label: '已收款', type: 'warning' },
  partial_shipped: { label: '部分发货', type: '' },
  shipped: { label: '已发货', type: 'warning' },
  partial_received: { label: '部分到货', type: '' },
  completed: { label: '已完成', type: 'success' },
  cancelled: { label: '已取消', type: 'danger' },
}

export const PAY_STATUS_MAP: Record<string, string> = {
  unpaid: '未收款',
  partial: '部分收款',
  paid: '已收款',
}

export const FULFILLMENT_TYPE_MAP: Record<string, string> = {
  wholesale: '批发',
  dropship: '分销直发',
}

export async function fetchDistOrders(params: {
  status?: string
  /** 多状态，逗号分隔，优先于 status */
  statuses?: string
  /** 收款状态，逗号分隔 unpaid|partial|paid */
  payStatus?: string
  /** 排除状态，逗号分隔 */
  excludeStatuses?: string
  fulfillmentType?: string
  distributorId?: number
  refSoId?: number
  refTraceId?: string
  keyword?: string
  createdAtStart?: string
  createdAtEnd?: string
  orderedAtStart?: string
  orderedAtEnd?: string
  sortBy?: string
  sortOrder?: string
  page?: number
  pageSize?: number
} = {}) {
  const res = await client.get('/dist-orders', { params })
  return unwrap<PageData<DistOrderListItem>>(res)
}

export async function fetchDistOrder(id: number) {
  return unwrap<DistOrder>(await client.get(`/dist-orders/${id}`))
}

export async function createDistOrder(data: DistOrderInput) {
  return unwrap<DistOrder>(await client.post('/dist-orders', data))
}

export async function updateDistOrder(id: number, data: DistOrderInput) {
  return unwrap<DistOrder>(await client.put(`/dist-orders/${id}`, data))
}

export async function deleteDistOrder(id: number) {
  return unwrap(await client.delete(`/dist-orders/${id}`))
}

export async function updateDistOrderItemPrices(
  id: number,
  items: { itemId: number; unitPrice: number }[],
) {
  return unwrap<DistOrder>(await client.put(`/dist-orders/${id}/item-prices`, { items }))
}

export async function submitDistOrder(id: number) {
  return unwrap<DistOrder>(await client.post(`/dist-orders/${id}/submit`))
}

export async function markDistOrderPaid(id: number) {
  return unwrap<DistOrder>(await client.post(`/dist-orders/${id}/mark-paid`))
}

export async function completeDistOrder(id: number) {
  return unwrap<DistOrder>(await client.post(`/dist-orders/${id}/complete`))
}

export async function cancelDistOrder(id: number) {
  return unwrap<DistOrder>(await client.post(`/dist-orders/${id}/cancel`))
}

export async function detachSalesOrder(data: {
  distNo: string
  orderNo?: string
  soId?: number
  reason?: string
}) {
  return unwrap<{ distOrder: DistOrder; unlinkWarning?: string }>(
    await client.post('/dist-orders/detach-sales-order', data),
  )
}

export async function mergeDistOrders(data: { sourceDistOrderIds: number[]; targetDistOrderId?: number }) {
  return unwrap<DistOrder & { mergedFromDistNos?: string[]; relinked?: number }>(
    await client.post('/dist-orders/merge', data),
  )
}
